src/v2/stack_aggregator.py

```diff
@@ -351,13 +358,169 @@ def execute(request: Dict, persist=True):
+            bindings['packages'] = list(packages)
+            started_at = time.time()
+            result = post_gremlin(query, bindings)
+            logger.info(
+                '%s took %0.2f secs for post_gremlin() batch request',
+                self._request.external_request_id, time.time() - started_at)
+            if result:
+                pkgs_with_vuln['result']['data'] += result['result']['data']
+
+        logger.info('%s took %0.2f secs for %s'
+                    'for total_results %d', self._request.external_request_id,
+                    time.time() - time_start, caller, len(pkgs_with_vuln['result']['data']))
+        return pkgs_with_vuln
+
+    def _filter_vulnerable_packages(self, vulnerabilities: List) -> Dict:
+        """Filter out vulnerabilities whose commit sha is out of vuln_commit_rules."""
+        logger.info('Executing filter_vulnerable_packages')
+
+        filter_vulnerabilities = defaultdict(list)
+        gh = GithubUtils()
+        for vuln in vulnerabilities:
+            package_name = vuln.get('package_name', [None])[0]
+            vuln_rules = vuln.get('vuln_commit_date_rules', [None])[0]
+            pseudo_version = self._normalized_packages.version_map.get(package_name)
+            if not pseudo_version:
+                logger.debug("Not a Pseudo Version.")
+                continue
+            time_stamp = gh.extract_timestamp(pseudo_version)
+            if all([vuln_rules, time_stamp,
+                    gh._is_commit_date_in_vuln_range(time_stamp, vuln_rules)]):
+                filter_vulnerabilities[package_name].append(vuln)
+        return filter_vulnerabilities
+
+    def _get_golang_package_details(self, pkg_node) -> Tuple[Package, PackageDetails]:
+        """Get Pseudo Golang Package Details."""
+        pkg_name = pkg_node.get('name', [None])[0]
+        ecosystem = pkg_node.get('ecosystem', [''])[0]
+        pkg = Package(name=pkg_name, version=self._normalized_packages.version_map[pkg_name])
+        latest_version = pkg_node.get('latest_version', [''])[0]
+        public_vulns, private_vulns = self._get_vulnerabilities(
+            self.filtered_vul.get(pkg_name, []))
+        recommended_latest_version = pkg_node.get("latest_non_cve_version", [""])[0]
+        pkg_details = PackageDataWithVulnerabilities(
+            **pkg.dict(),
+            ecosystem=ecosystem,
+            latest_version=latest_version,
+            github={},
+            licenses=[],
+            url=_get_snyk_package_link(ecosystem, pkg_name),
+            private_vulnerabilities=private_vulns,
+            public_vulnerabilities=public_vulns,
+            recommended_version=recommended_latest_version)
+
+        return pkg, pkg_details
+
+    def _get_package_details_from_graph_for_pseudo_versions(self) -> List:
+        """Stack analyses call only for pseudo version applicable for golang."""
+        logger.debug('Executing get_batch_sa_data_for_pseudo_version')
+        get_modules_query = """
```

Comment:
This query is very heavy as it fetches all vulnerable packages for modules, can you run gremlin query with some sample data and check response time and size. I have modified this query in CA as https://github.com/fabric8-analytics/fabric8-analytics-server/pull/700/files#diff-5c06ff88b26356bd63c9cadbce799b3814d67d12700531431cbb499d0f2b68fcR61, PTAL.
Basically I am filtering based on package which are requested by API, this reduces data fetched from gremlin.

Reply:
> This query is very heavy
I believe Querying on Packages is comparitevely heavier.

Compared to CA, SA is Async. 
We can take it as a a refactor job if latency is concerning and SLO's are not met. 
Not a concern as of now.

Reply:
Can you check this query with few sample SA call on popular golang repos? I suspect you will get timeout from gremlin, this code will not execute at all. Your trying to fetch all modules and their package vulnerabilities in once gremlin call.

I strongly suggest to consider filtering only based on packages that are requested SA request. This will definitely be a very small sub-set of data than above module based query.

Reply:
the list passed will contain only modules, as per our prev conversation. If the number of modules exceed 50, it can be easily batched. whats the problem here?

Reply:
@yzainee  This query will filter all nodes for which module_name matches, which means it will return module + all packages under that module which are vulnerable.

Reply:
@deepak1725 I am still not confident that the module query will work for repo with day 50-80 module deps. Can you please share time it takes to execute this query for https://github.com/redhat-developer/helm which as ~75 modules.

Reply:
@dgpatelgit I think you are not getting what I and Yusuf are trying to convey here.

1. Querying in SA is on the batch, based on `GREMLIN_QUERY_SIZE` (default 50)
So, even in the case of Your example Repo of Helm having 76 packages (query to gremlin will be in 2 parts: 50 + 26)

2. In Case of Packages-> Search is like Select n out of Billions of Packages 
In Case of Modules -> Search is like Select n out of Millions of Modules.
PS: Billion > Million
I hope you get the idea, which will be faster.

Reply:
the total number of vulnerabilities in golang is around 300. I dont think it would be a big search operation anyways

Reply:
here he will be sending only the module which have a pseudo version. are you expecting 80-100 deps with pseudo version in a single file?

Reply:
exactly Yusuf. Even i don't feel any concern here.